rules/no-single-promise-in-promise-methods.js

```diff
@@ -0,0 +1,49 @@
+'use strict';
+const MESSAGE_ID = 'no-single-promise-in-promise-methods';
+const messages = {
+	[MESSAGE_ID]: 'Promise.{{ method }} contains an array of a single element.',
+};
+
+const isPromiseMethodWithSinglePromise = node =>
+	node.callee.type === 'MemberExpression'
+	&& node.callee.object.type === 'Identifier'
+	&& node.callee.object.name === 'Promise'
+	&& node.callee.property.type === 'Identifier'
+	&& ['all', 'allSettled', 'any', 'race'].includes(node.callee.property.name)
```

Comment:
This part could be reused with your other rule. Maybe move it to a separate method and move it to https://github.com/sindresorhus/eslint-plugin-unicorn/tree/main/rules/utils